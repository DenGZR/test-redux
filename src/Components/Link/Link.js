import React, { Component } from 'react';

const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'my1w-normals',
};

const styleLink = {
  color: 'red',
  textTransform: 'uppercase'
};

export default class Link extends Component {
  constructor() {
    super();

    this._onMouseEnter = this._onMouseEnter.bind(this);
    this._onMouseLeave = this._onMouseLeave.bind(this);

    this.state = {
      class: STATUS.NORMAL,
    };
  }

  _onMouseEnter() {
    this.setState({class: STATUS.HOVERED});
  }

  _onMouseLeave() {
    this.setState({class: STATUS.NORMAL});
  }

  render() {
    return (
      <a
        style={styleLink}
        className={this.state.class}
        href={this.props.page || '/'}
        onMouseEnter={this._onMouseEnter}
        onMouseLeave={this._onMouseLeave}
      >
        {this.props.children}
      </a>
    );
  }
}
