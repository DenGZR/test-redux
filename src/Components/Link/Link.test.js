import React from 'react';
import Link from './Link';
import renderer from 'react-test-renderer';

describe('<Link />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Link page="http://www.facebook.com">Facebook</Link>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders as an anchor when no page is set', () => {
    const tree = renderer.create(<Link>Facebook</Link>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('properly escapes quotes', () => {
    const tree = renderer
      .create(<Link>{"\"Facebook\" \\'is \\ 'awesome'"}</Link>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('changes the class when hovered', () => {
    const component = renderer.create(
      <Link page="http://www.facebook.com">Facebook</Link>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseEnter();
    // console.log('tree --> ', tree);
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();
    // console.log('tree --> ', tree);
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
