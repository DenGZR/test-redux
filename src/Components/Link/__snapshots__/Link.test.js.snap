// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> changes the class when hovered 1`] = `
<a
  className="my-normals"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "color": "red",
      "textTransform": "uppercase",
    }
  }
>
  Facebook
</a>
`;

exports[`<Link /> changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "color": "red",
      "textTransform": "uppercase",
    }
  }
>
  Facebook
</a>
`;

exports[`<Link /> changes the class when hovered 3`] = `
<a
  className="my-normals"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "color": "red",
      "textTransform": "uppercase",
    }
  }
>
  Facebook
</a>
`;

exports[`<Link /> properly escapes quotes 1`] = `
<a
  className="my-normals"
  href="/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "color": "red",
      "textTransform": "uppercase",
    }
  }
>
  "Facebook" \\'is \\ 'awesome'
</a>
`;

exports[`<Link /> renders as an anchor when no page is set 1`] = `
<a
  className="my-normals"
  href="/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "color": "red",
      "textTransform": "uppercase",
    }
  }
>
  Facebook
</a>
`;

exports[`<Link /> renders correctly 1`] = `
<a
  className="my-normals"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "color": "red",
      "textTransform": "uppercase",
    }
  }
>
  Facebook
</a>
`;
